tests/test_utils.py

```diff
@@ -0,0 +1,100 @@
+"""Test utils."""
+import io
+import os
+import json
+import pandas as pd
+import tempfile
+import unittest
+from bitarray import bitarray
+from clkhash.serialization import serialize_bitarray
+from clkhash.clk import generate_clk_from_csv
+from clkhash.schema import from_json_file
+from blocklib import generate_candidate_blocks
+from anonlinkclient.utils import deserialize_filters, generate_candidate_blocks_from_csv, combine_clks_blocks
+
+from tests import *
+
+
+class TestUtils(unittest.TestCase):
+
+    def test_deserialize_bitarray(self):
+        """Test deserialize bitarray."""
+        # create simple bitarray bloom filter
+        lst = [True, False, True, False] * 2
+        bf = bitarray(lst)
+        serialized_bf = serialize_bitarray(bf)
+        deserialized_bf = deserialize_filters([serialized_bf])
+        assert deserialized_bf == [bf]
+
+    def test_generate_candidate_blocks_from_csv_exception(self):
+        """Test exception case for generate candidate blocks from csv."""
+        data = [[1, 'Joyce'],
+                [2, 'Fred']]
+        input_f = io.StringIO(str(data))
+        schema_f = open(os.path.join(TESTDATA, 'bad-schema-v1.json'))
+
+        # test invalid header
+        header = 'Header'
+        with self.assertRaises(ValueError):
+            generate_candidate_blocks_from_csv(input_f, schema_f, header)
+
+    def test_combine_clks_blocks(self):
```

Comment:
This is a beast of a test - it would be helpful to document a bit more what it is doing and what it is trying to test. 

I suggest trying to simplify, perhaps by including example files for blocks and clks as test data (similar to 'dirty_1000_50_1.csv') or extract the setup into a helper function.